ecs/pod_creator_test.go

```diff
@@ -1,12 +1,286 @@
+			require.NoError(t, err)
+
+			p, err := podCreator.CreatePod(ctx, opts)
+			require.Error(t, err)
+			require.Zero(t, p)
+		},
+		"CreatePodFailsWithSecretsNoVault": func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault) {
+			envVar := cocoa.NewEnvironmentVariable().SetName("evName").
+				SetSecretOptions(*cocoa.NewSecretOptions().SetName("name").SetValue("value"))
+			containerDef := cocoa.NewECSContainerDefinition().SetImage("image").
+				SetEnvironmentVariables([]cocoa.EnvironmentVariable{*envVar})
+			require.NotNil(t, containerDef.EnvVars)
+
+			execOpts := cocoa.NewECSPodExecutionOptions().SetCluster(testutil.ECSClusterName())
+			assert.NoError(t, execOpts.Validate())
+
+			opts := cocoa.NewECSPodCreationOptions().SetName("name").
+				AddContainerDefinitions(*containerDef).
+				SetMemoryMB(128).
+				SetCPU(128).
+				SetTaskRole("role").
+				AddTags("tag").
+				SetExecutionOptions(*execOpts)
+			require.NotZero(t, opts)
+			assert.NotZero(t, *opts)
+			assert.NoError(t, opts.Validate())
+
+			podCreator, err := NewBasicECSPodCreator(c, nil)
+			require.NoError(t, err)
+
+			p, err := podCreator.CreatePod(ctx, opts)
+			require.Error(t, err)
+			require.Zero(t, p)
+		},
+		"CreatePodFailsWithIncompleteContainerDef": func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault) {
+			envVar := cocoa.NewEnvironmentVariable().SetName("ev").
+				SetSecretOptions(*cocoa.NewSecretOptions().SetName(testutil.NewSecretName("name")).SetValue("value"))
+			containerDef := cocoa.NewECSContainerDefinition().SetImage("image").
+				SetEnvironmentVariables([]cocoa.EnvironmentVariable{*envVar}).
+				SetName("container")
+			require.NotNil(t, containerDef.EnvVars)
+
+			execOpts := cocoa.NewECSPodExecutionOptions().SetCluster(testutil.ECSClusterName())
+			assert.NoError(t, execOpts.Validate())
+
+			opts := cocoa.NewECSPodCreationOptions().SetName("name").
+				AddContainerDefinitions(*containerDef).
+				SetMemoryMB(128).
+				SetCPU(128).
+				SetTaskRole("role").
+				AddTags("tag").
+				SetExecutionOptions(*execOpts)
+			require.NotZero(t, opts)
+			assert.NotZero(t, *opts)
+			assert.NoError(t, opts.Validate())
+
+			podCreator, err := NewBasicECSPodCreator(c, v)
+			require.NoError(t, err)
+
+			p, err := podCreator.CreatePod(ctx, opts)
+			require.Error(t, err)
+			require.Zero(t, p)
+		},
+		"CreatePodFailsWithInvalidSecretName": func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault) {
+			envVar := cocoa.NewEnvironmentVariable().SetName("ev").
+				SetSecretOptions(*cocoa.NewSecretOptions().SetName("name").SetValue("value"))
+			containerDef := cocoa.NewECSContainerDefinition().SetImage("image").
+				SetEnvironmentVariables([]cocoa.EnvironmentVariable{*envVar}).
+				SetName("container").
+				SetMemoryMB(128).
+				SetCPU(128)
+			require.NotNil(t, containerDef.EnvVars)
+
+			execOpts := cocoa.NewECSPodExecutionOptions().SetCluster(testutil.ECSClusterName())
```

Comment:
This test should set the execution role to ensure that the test doesn't just fail because the execution role is missing.